File: problemas/iniciante/1114/1114.go
package main

import (
    "fmt"
)

func main() {
    var senha int64

    fmt.Scanf("%d", &senha)
    for {
        if senha == 2002 {
            fmt.Println("Acesso Permitido")
            break
        }
        fmt.Println("Senha Invalida")
        fmt.Scanf("%d", &senha)
    }
}

File: problemas/iniciante/1114/1114.hs
import Text.Printf

senha :: Int -> IO ()
senha 2002 = printf "Acesso Permitido\n"
senha n = do
  printf "Senha Invalida\n"
  numero <- readLn :: IO Int
  senha numero

main :: IO ()
main = do
  numero <- readLn :: IO Int
  senha numero